### DISCUSSION TITLE
Add option to pass incoming Clock to Output(`s)

### DISCUSSION THREAD

**Discussion by patriciachadwick62** [SynthstromAudible/DelugeFirmware#1403] [Ideas]
### Please describe the problem:

incoming Clock in / MIDI Clock isn`t send to Output(`s)
no matter if Clock is  in Midi device / output setting enabled

### What is the expected behavior?

If clock out is enabled, the Clock should be send there
no matter if received or from deluge itself

### What hardware did you reproduce it with?

OLED

### What firmware did you reproduce it with?

Release 1.0/Amadeus, Nightly, Built from community, other

### What is the firmware name:

all vers.

**Comment by m-m-adams** [comment]
Hey this is a deliberate choice - passing through clock can end up with doubled clocks in lots of situations. I'm going to move this to discussions as a feature request to add a toggle to pass clock through

**Comment by manysounds** [comment]
I suggested this a while ago.  
Having clock pass-through separated from MIDI Thru would be awesome.

**Comment by patriciachadwick62** [comment]
**Conclusion
Would be nice 
if the now **_”enabled”_** would be renamed as **“_enabled as master_”** (stay standard setting [to prevent the doubleclock problem])                     **in MIDI/ Device/…/Clock/output**/
and **ad** a **new** option now called **“enabled”**
(with the setting to output always clock to devices/output’s set so
  [normaly the internal and when slave the clock thru])**

**So wouldn’t it best put this to a pull request
Thank Y**





Thinking
Ok, lets discuss if this is an issue

Thought this is an issue,
when somthing set to send ( MIDI/ Device/…/Clock/output/enabled)
and its not send.   

   there is no extra toggel needed    (only a logical when…)
when set to output it should do so 
if didn’t should it can be disabled
(and the double clock shouldn’t happen if the internal is not added what isn’t)

its right the clock outputed when slave, should be to thru the  incoming clock
(and not the internal think used in this situation to sync scale if..)
Think thats the normal handling, processing can’t make the signal better


Think this issue hasn’t isn’t a problem in most situations the clock Deluge receives 
is then given directly to other Devices too
but if you want to reduce the setup and things to  carry as much as possible 
this way to use is a option
(So to use this way can let fall away a splitter and a few cables and shorten a few)
